common/src/bitset.rs

```diff
@@ -247,7 +246,22 @@ impl BitSet {
         }
     }
 
+    /// Intersect with serialized bitset
+    pub fn intersect_with(&mut self, other: &ReadSerializedBitSet) {
```

Comment:
Rename `intersect_update`...

These set operation typically mutate or not mutate (in this PR you introduced both kind.)
It's a good idea to make it explicit in the method name that this is a mutating version.